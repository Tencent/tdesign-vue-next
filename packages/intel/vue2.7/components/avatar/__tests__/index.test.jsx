import { mount } from '@vue/test-utils';
import Avatar, { AvatarGroup } from '@/src/avatar/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Avatar', () => {
  // test props api
  describe(':props', () => {
    it(':image', () => {
      const wrapper = mount({
        render() {
          return <Avatar image="https://tdesign.gtimg.com/site/avatar.jpg"></Avatar>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':icon', () => {
      const wrapper = mount({
        render() {
          return <div></div>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':shape', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <Avatar shape="round">W</Avatar>
              <Avatar shape="circle">W</Avatar>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':size', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <Avatar size="small">Avatar</Avatar>
              <Avatar size="medium">W</Avatar>
              <Avatar size="large">W</Avatar>
              <Avatar size="80px">Avatar</Avatar>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('<slot>', () => {
    it('<icon>', () => {
      const wrapper = mount(Avatar, {
        scopedSlots: {
          icon: '<div></div>',
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});

describe('AvatarGroup', () => {
  describe(':props', () => {
    it(':cascading', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <AvatarGroup cascading="right-up">
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
              <AvatarGroup cascading="left-up">
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':collapseAvatar', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <AvatarGroup cascading="right-up">
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
              <AvatarGroup cascading="left-up">
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':max', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <AvatarGroup max={2}>
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':placement', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <AvatarGroup max={2} placement="bottom">
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':popupProps', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <AvatarGroup max={2} popupProps={{ trigger: 'click', placement: 'bottom' }}>
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':size', () => {
      const wrapper = mount({
        render() {
          return (
            <div>
              <AvatarGroup size="small">
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
                <Avatar>W</Avatar>
              </AvatarGroup>
            </div>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
