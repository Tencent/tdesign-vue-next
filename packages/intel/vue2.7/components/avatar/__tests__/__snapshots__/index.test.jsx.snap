// Vitest Snapshot v1

exports[`Avatar > :props > :icon 1`] = `<div />`;

exports[`Avatar > :props > :image 1`] = `
<div
  class="t-avatar t-avatar--circle"
>
  <div
    class="t-image__wrapper t-image__wrapper--shape-square"
  >
    <img
      alt=""
      class="t-image t-image--fit-fill t-image--position-center"
      src="https://tdesign.gtimg.com/site/avatar.jpg"
    />
    <div
      class="t-image__loading"
    >
       
    </div>
  </div>
</div>
`;

exports[`Avatar > :props > :shape 1`] = `
<div>
  <div
    class="t-avatar t-avatar--round"
  >
    <span>
      W
    </span>
  </div>
  <div
    class="t-avatar t-avatar--circle"
  >
    <span>
      W
    </span>
  </div>
</div>
`;

exports[`Avatar > :props > :size 1`] = `
<div>
  <div
    class="t-avatar t-size-s t-avatar--circle"
  >
    <span>
      Avatar
    </span>
  </div>
  <div
    class="t-avatar t-size-m t-avatar--circle"
  >
    <span>
      W
    </span>
  </div>
  <div
    class="t-avatar t-size-l t-avatar--circle"
  >
    <span>
      W
    </span>
  </div>
  <div
    class="t-avatar t-avatar--circle"
    style="width: 80px; height: 80px; font-size: 40px;"
  >
    <span>
      Avatar
    </span>
  </div>
</div>
`;

exports[`Avatar > <slot> > <icon> 1`] = `
<div
  class="t-avatar t-avatar--circle t-avatar__icon"
>
  <div />
</div>
`;

exports[`AvatarGroup > :props > :cascading 1`] = `
<div>
  <div
    class="t-avatar-group t-avatar--offset-right"
  >
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
  </div>
  <div
    class="t-avatar-group t-avatar--offset-left"
  >
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup > :props > :collapseAvatar 1`] = `
<div>
  <div
    class="t-avatar-group t-avatar--offset-right"
  >
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
  </div>
  <div
    class="t-avatar-group t-avatar--offset-left"
  >
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup > :props > :max 1`] = `
<div>
  <div
    class="t-avatar-group t-avatar--offset-right"
  >
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle t-avatar__collapse"
    >
      <span>
        +1
      </span>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup > :props > :placement 1`] = `
<div>
  <div
    class="t-avatar-group t-avatar--offset-right"
    placement="bottom"
  >
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle t-avatar__collapse"
    >
      <span>
        +1
      </span>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup > :props > :popupProps 1`] = `
<div>
  <div
    class="t-avatar-group t-avatar--offset-right"
  >
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-m t-avatar--circle t-avatar__collapse"
    >
      <span>
        +1
      </span>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup > :props > :size 1`] = `
<div>
  <div
    class="t-avatar-group t-avatar--offset-right"
  >
    <div
      class="t-avatar t-size-s t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-s t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
    <div
      class="t-avatar t-size-s t-avatar--circle"
    >
      <span>
        W
      </span>
    </div>
  </div>
</div>
`;
